research_daps/orms/glass.py

```diff
@@ -0,0 +1,202 @@
+# type: ignore
+"""Glass AI ORMs
+
+Assumptions:
+- Organisation names and URL's do not change
+- Sectors do change [TODO]
+- Addresses do change [TODO]
+
+"""
+from sqlalchemy import Table, Column, Integer, String, ForeignKey, Date, Boolean
+from sqlalchemy.orm import relationship, backref
+from research_daps import declarative_base
+
+Base = declarative_base()
+
+# organisation_address = Table(
+#     "organisation_address",
+#     Base.metadata,
+#     Column("org_id", Integer, ForeignKey("organisation.org_id")),
+#     Column("address_id", Integer, ForeignKey("address.address_id")),
+#     # TODO: Active? boolean column indicating whether address is in latest data dump
+#     # TODO: Rank Integer column indicating address rank
+#     # TODO: Date Date column indicating date of latest dump with this relationship
+# )
+class OrganisationAddress(Base):
+    """Association object between organisations and addresses"""
+
+    org_id = Column(Integer, ForeignKey("organisation.org_id"), primary_key=True)
+    address_id = Column(Integer, ForeignKey("address.address_id"), primary_key=True)
+    # TODO: do we really want to keep ranks? If ranks constantly change then
```

Comment:
regarding this TODO: is the intention that you a) upsert in new collections or b) create duplicate entries in order to create a longitudinal dataset? If the former, then the ranks changing doesn't make a difference to us. If the latter, then you will also need to add `date` to the primary key (but keep the index as well), and then the proliferation of data shouldn't make too much difference since slicing on the date will be quick (unless you're worried about diskspace, but don't)